In [23]:
import sys
import os
import logging
 
# Add the project root directory to Python path
sys.path.append(os.path.abspath(os.path.join(os.path.dirname('__file__'), '..')))
%reload_ext autoreload
%autoreload 2 


In [ ]:
import pandas as pd
import oracledb
from scripts.database_setup import create_connection, create_tables, insert_banks, insert_reviews

# Loading the datasets pathes
REVIEWS_PATH = '../data/analyzed_reviews.csv'
CBE_PATH = '../data/processed/cbe_review.csv'
BOE_PATH = '../data/processed/boe_review.csv'
DASHEN_PATH = '../data/processed/dashen_review.csv'

# Main function to execute the database operations."""
connection = None
try:
    connection = create_connection()
    create_tables(connection)
    insert_banks(connection)
    insert_reviews(connection, CBE_PATH, 'CBE')
    insert_reviews(connection, BOE_PATH, 'BOE')
    insert_reviews(connection, DASHEN_PATH, 'Dashen')
except Exception as e:
    print(f"An error occurred: {str(e)}")
finally:
    if connection:
        connection.close()
        print("Database connection closed.")



Database connection established.
Banks table created successfully.
Reviews table created successfully.
Inserted 3 banks into the database.
Inserted 600 reviews for CBE into the database.
Inserted 596 reviews for BOE into the database.
Inserted 408 reviews for Dashen into the database.
Database connection closed.


## Output Analysis
  - It looks like the output from my database operations indicates a successful workflow. 

1. Database Connection: The connection to the database was established successfully, which is a positive start
2. Table Creation:
   * The Banks table was created successfully.
   * The Reviews table was also created successfully.
3. Data Insertion:Inserted 3 banks into the database, suggesting that the Banks table now has entries.
Inserted a total of 1,604 reviews across different banks:
   * 600 reviews for CBE.
   * 596 reviews for BOE.
   * 408 reviews for Dashen.
4. Connection Closed: The database connection was closed properly, indicating that all operations were completed without errors.